presto-tests/src/test/java/io/prestosql/execution/EventsAwaitingQueries.java

```diff
@@ -43,7 +43,13 @@ MaterializedResult runQueryAndWaitForEvents(@Language("SQL") String sql, int num
         return runQueryAndWaitForEvents(sql, numEventsExpected, session, Optional.empty());
     }
 
-    MaterializedResult runQueryAndWaitForEvents(@Language("SQL") String sql, int numEventsExpected, Session session, Optional<String> expectedExceptionRegEx)
+    MaterializedResult runQueryAndWaitForEvents(@Language("SQL") String sql, int numEventsExpected, Session session, String expectedExceptionRegEx)
```

Comment:
perhaps it's even better when the exception is asserted in the calling code, as if using `org.assertj.core.api.Assertions.catchThrowable` + `Assertions.assertThat`

Reply:
Possibly. PTAL at current shape and we can make it a followup.